src/parser/ion_1_0/current_symbol_table.rs

```diff
@@ -65,37 +81,40 @@ impl CurrentSymbolTable {
 pub(crate) fn update_current_symbol_table(
     current: &mut CurrentSymbolTable,
     encountered: &Option<Struct>,
-) {
+) -> Result<(), SymbolError> {
     let (imports, symbols): (TableImport, Vec<SymbolToken>) = match encountered {
         None => (TableImport::None, vec![]),
-        Some(Struct { fields: values }) => {
-            let keys: HashMap<&str, usize> = values
-                .iter()
-                .enumerate()
-                .filter_map(|(i, val)| match &val.0 {
-                    SymbolToken::Known { text } => Some((text.as_str(), i)),
-                    SymbolToken::Unknown { .. } => None,
-                    SymbolToken::Zero => None,
-                })
-                .collect();
+        Some(Struct { fields }) => {
+            // Confirm there are no duplicate imports/etc.
+            let keys = safe_key_map(fields).map_err(|_| SymbolError::InvalidSymbolTable)?;
```

Comment:
Yeah, I definitely created this as a reaction to the tests that imply duplicate imports and symbols fields are an error. I'll make the change since the only fields I need to verify at the moment are imports and symbols.

For now I'll obey the tests but I'll drop a note in with a link to the issue. I'll also change safe_key_map into key_map which generates a HashMap<&str, Vec<usize>> so we can still have a utility for accessing fields where necessary (we do it in at least two separate places so I got a little fixated on DRY here, LMK if this works with your suggestions).

Reply:
Sticking with the test behavior seems like the right move, I wouldn't be too surprised if the Ion maintainers end up wanting to standardize the current behavior of the official implementations.